eox_nelp/external_certificates/tests/api/v1/test_views.py

```diff
@@ -0,0 +1,196 @@
+        """Test successful creation/update of an external certificate.
+        Expected behavior:
+            - The view should return a 201 CREATED response.
+            - The response should contain the certificate ID and URLs.
+        """
+        request_data = {
+            "user_id": self.user.id,
+            "course_id": self.course_id,
+            "certificate_response": {
+                "message": "Certificate created successfully",
+                "certificate_id": "TEST25YO59VV76QL",
+                "certificate_urls": {
+                    "en": "https://example.com/certificate-en.pdf",
+                    "ar": "https://example.com/certificate-ar.pdf",
+                },
+            },
+        }
+        expected_response_data = {
+            'id': 1,
+            'certificate_id': 'TEST25YO59VV76QL',
+            'certificate_url_en': 'https://example.com/certificate-en.pdf',
+            'certificate_url_ar': 'https://example.com/certificate-ar.pdf',
+            'created_at': '2025-03-12T12:24:33.759224',
+            'user': str(self.user.id),
+            'course_overview': self.course_id,
+        }
+
+        with self.assertLogs("eox_nelp.external_certificates.models", level="INFO"):
+            response = self.client.post(self.url, data=request_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        for key, value in expected_response_data.items():
+            if key != "created_at":
+                self.assertEqual(response.json()[key], value)
+
+    @patch.object(JwtHasScope, 'has_permission', return_value=False)
+    def test_upsert_external_certificate_wrong_permission(self, _):
+        """Test successful creation/update of an external certificate.
+        Expected behavior:
+            - The view should return a 403 bad permissions response.
+            - - Expected error response json.
+        """
+        request_data = {
+            "user_id": self.user.id,
+            "course_id": self.course_id,
+            "certificate_response": {
+                "message": "Certificate created successfully",
+                "certificate_id": "TEST25YO59VV76QL",
+                "certificate_urls": {
+                    "en": "https://example.com/certificate-en.pdf",
+                    "ar": "https://example.com/certificate-ar.pdf",
+                },
+            },
+        }
+        expected_response_data = {'detail': 'JWT missing required scopes.'}
+
+        response = self.client.post(self.url, data=request_data, format="json")
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertDictEqual(response.json(), expected_response_data)
+
+    @patch.object(JwtHasScope, 'has_permission', return_value=True)
+    def test_upsert_external_certificate_missing_user_id(self, _):
+        """Test missing required fields. user_id and result_notification_id are missing.
```

Comment:
the name just mentioned user_id could you make this general test_upsert_external_certificate_missing_key and use dtt